spec/components/search_spec.rb

```diff
@@ -1064,6 +1064,16 @@ def search
       expect(Search.execute('badge:"test"').posts.length).to eq(0)
     end
 
+    it 'can match exact phrases' do
+      post = Fabricate(:post, raw: 'this is a test post with a URL https://some.site.com/search?q=test.test.test some random text I have to add')
+      post2 = Fabricate(:post, raw: 'test URL post with')
+
+      expect(Search.execute("test post with a URL).posts").posts).to eq([post2, post])
+      expect(Search.execute("\"test post with a URL\"").posts).to eq([post])
+      expect(Search.execute("\"https://some.site.com/search?q=test.test.test\"").posts).to eq([post])
+      expect(Search.execute("\" with a URL https://some.site.com/search?q=test.test.test\"").posts).to eq([post])
```

Comment:
We should probably use another string delimiter 😉 

```suggestion
      expect(Search.execute('"test post with a URL"').posts).to eq([post])
      expect(Search.execute('"https://some.site.com/search?q=test.test.test"').posts).to eq([post])
      expect(Search.execute('" with a URL https://some.site.com/search?q=test.test.test"').posts).to eq([post])
```

or, the more "in-the-face"

```suggestion
      expect(Search.execute(%{"test post with a URL"}).posts).to eq([post])
      expect(Search.execute(%{"https://some.site.com/search?q=test.test.test"}).posts).to eq([post])
      expect(Search.execute(%{" with a URL https://some.site.com/search?q=test.test.test"}).posts).to eq([post])
```